contracts/zero-ex/contracts/src/vendor/ILiquidityProvider.sol

```diff
@@ -0,0 +1,90 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+
+interface ILiquidityProvider {
+
+    /// @dev Trades `takerToken` for `makerToken`. The amount of `takerToken`
+    ///      to sell must be transferred to the contract prior to calling this
+    ///      function to trigger the trade.
+    /// @param takerToken The token being sold.
+    /// @param makerToken The token being bought.
+    /// @param recipient The recipient of the bought tokens.
+    /// @param minBuyAmount The minimum acceptable amount of `makerToken` to buy.
+    /// @param auxiliaryData Arbitrary auxiliary data supplied to the contract.
+    /// @return boughtAmount The amount of `makerToken` bought.
+    function sellTokenForToken(
+        address takerToken,
+        address makerToken,
+        address recipient,
+        uint256 minBuyAmount,
+        bytes calldata auxiliaryData
+    )
+        external
+        returns (uint256 boughtAmount);
+
+    /// @dev Trades ETH for token. ETH must be sent to the contract prior to
+    ///      calling this function to trigger the trade.
+    /// @param makerToken The token being bought.
+    /// @param recipient The recipient of the bought tokens.
+    /// @param minBuyAmount The minimum acceptable amount of `makerToken` to buy.
+    /// @param auxiliaryData Arbitrary auxiliary data supplied to the contract.
+    /// @return boughtAmount The amount of `makerToken` bought.
+    function sellEthForToken(
+        address makerToken,
+        address recipient,
+        uint256 minBuyAmount,
+        bytes calldata auxiliaryData
+    )
+        external
+        returns (uint256 boughtAmount);
```

Comment:
Isn't it cheaper to just make this `payable` than having to do a separate transfer call?

Reply:
it's actually a bit more expensive because we have to make 2 payable calls (EP => sandbox => liquidity provider).
47194 gas vs 53008 according to ganache

Reply:
Ah, forgot about the sandbox. Maybe it should be payable anyway if we ever interact with it from a transformer?

Reply:
yeah I suppose so